scripts/leadtime.ts

```diff
@@ -118,6 +81,50 @@ const script: Script = async (config: Config, argv: Argv) => {
     } else {
         console.error("Tickets were not fetched properly :/");
     }
-};
+}
+
+namespace script {
+    function prettyPrintDate(date: Date): string {
+        return `${date.getFullYear()}-${date.getMonth() + 1}-${date.getDate()}`;
+    }
+
+    function prettyPrintTimes(values: { [key: string]: number }, statuses: string[]): string {
+        return statuses
+            .map(s => s.toLowerCase())
+            .map(s => values[s] || 0)
+            .join(",");
+    }
+
+    export function getIssueTimeStrings<IssueWithChangelog extends Issue & HasChangelog>(
+        issues: IssueWithChangelog[],
+        statuses: string[],
+        finalStatuses: string[],
+        showSummary: boolean
+    ): string[] {
+        const headings = [];
+        headings.push("Key");
+        if (showSummary) headings.push("Summary");
+        headings.push("Created");
+        headings.push("Finished");
+        headings.push(...statuses);
+        const titleLine = headings.join(",");
+
+        const infoResults = issues.map(issue => Jira.getIssueTimings(issue, finalStatuses));
+
+        const lines = infoResults.map(info => {
+            const row = [];
+            row.push(info.key);
+            if (showSummary) row.push(`"${info.summary.replace('"', '\\"')}"`);
+            row.push(prettyPrintDate(info.created));
+            row.push(info.finished ? prettyPrintDate(info.finished) : "");
+            row.push(prettyPrintTimes(info.times, statuses));
+            return row.join(",");
+        });
+
+        return [titleLine].concat(lines);
+    }
+}
+
+const thisRaisesACompileTimeErrorIfScriptIsMalformatted: Script = script;
```

Comment:
I like this row. It's very clear what this row does even for TypeScript newbie like me. :+1: